app/src/androidTest/java/ch/epfl/sweng/studdybuddy/SettingsFragmentTest.java

```diff
@@ -109,15 +103,29 @@ public void run() {
             }
         });
         onView(withId(R.id.text_location_set_up)).check(matches(withText("Default Location: : ")));
-
-        fragment.getActivity().finish();
-
-
     }
 
+    @Test
+    public void signoutButtonTest() throws InterruptedException {
+        onView(withId(R.id.btn_sign_out)).perform(click());
+        Thread.sleep(1000);
+        onView(withId(R.id.googleBtn)).check(matches(isDisplayed()));
```

Comment:
As you use an IntentTestRule instance, it would be better to use this : 
```suggestion
        intended(hasComponent(GoogleSignInActivity.class.getName());
```
or DummyGoogleSignInActivity if you use a dummy button

I think that checking if the child activity's button is displayed is necessary if and only if you are forced to use an ActivityTestRule

Reply:
I inititially used the intended you're suggesting but it failed for no reason. I will try again and see if it passes with travis.